MaxText/max_utils.py

```diff
@@ -333,6 +336,10 @@ def create_device_mesh(config, devices=None):
   if multi_slice_env:
     dcn_parallelism = fill_unspecified_mesh_axes(dcn_parallelism, num_slices, "DCN")
     mesh = mesh_utils.create_hybrid_device_mesh(ici_parallelism, dcn_parallelism, devices)
+  elif config.model_name == "llama2-70b" and config.sharding_strategy == "experimental":
```

Comment:
Same here, I thought this looks like llama2-70b specific the code is for multi-host inference. We should just add a `config.multi_host_inference` key.

Reply:
Could we do it later? Even with `kv_proj_kernel_axes `, we still need to use `if` `else`, unless we make it general to all models which need further testing. 

Simply moving it to config doesn't work.  I only want to focus on the llama2 70B optimization in this PR.

Reply:
Could we add a boolean "allow_split_physical_axes" to the base.yml config instead of using

```
elif config.model_name == "llama2-70b" and config.sharding_strategy == "experimental":
```

Reply:
done.